pliers/extractors/audio.py

```diff
@@ -488,4 +495,126 @@ class PercussiveExtractor(LibrosaFeatureExtractor):
+
+        MODULE_PATH = path.dirname(self.yamnet.__file__)
+        LABELS_PATH = path.join(MODULE_PATH, 'yamnet_class_map.csv')
+        self.weights_path = weights_path or path.join(MODULE_PATH, 'yamnet.h5')
+        self.hop_size = hop_size
+        self.yamnet_kwargs = yamnet_kwargs or {}
+        self.params = self.yamnet.params
+        self.params.PATCH_HOP_SECONDS = hop_size
+        for par, v in self.yamnet_kwargs.items():
+            setattr(self.params, par, v)
+        if self.params.PATCH_WINDOW_SECONDS != 0.96:
+            logging.warning('Custom values for PATCH_WINDOW_SECONDS were '
+                'passed. YAMNet was trained on windows of 0.96s. Different '
+                'values might yield unreliable results.')
+
+        self.top_n = top_n
+        all_labels = pd.read_csv(LABELS_PATH)['display_name'].tolist()
+        if labels is not None:
+            missing = list(set(labels) - set(all_labels))
+            labels = list(set(labels) & set(all_labels))
+            if missing:
+                logging.warning(f'Labels {missing} do not exist. Dropping.')
+            self.labels = labels
+            self.label_idx = [i for i, l in enumerate(all_labels) 
+                              if l in labels]
+        else:
+            self.labels = all_labels
+            self.label_idx = range(len(all_labels))
+        super(AudiosetLabelExtractor, self).__init__()
+
+    def _extract(self, stim):
+        self.params.SAMPLE_RATE = stim.sampling_rate
+
+        if self.params.SAMPLE_RATE >= 2 * self.params.MEL_MAX_HZ:
+            if self.params.SAMPLE_RATE != 16000:
+                logging.warning(
+                    'The sampling rate of the stimulus is '
+                    f'{self.params.SAMPLE_RATE}Hz. YAMNet was trained on '
+                    ' audio sampled at 16000Hz. This should not impact '
+                    'predictions, but you can resample the input using '
+                    'AudioResamplingFilter for full conformity ' 
+                    'to training.')
+            if self.params.MEL_MIN_HZ != 125 or self.params.MEL_MAX_HZ != 7500:
+                logging.warning(
+                    'Custom values for MEL_MIN_HZ and MEL_MAX_HZ '
+                    'were passed. Changing these defaults might affect '
+                    'model performance.')
+        else:
+            raise ValueError(
+                'The sampling rate of your stimulus '
+                f'({self.params.SAMPLE_RATE}Hz) must be at least twice the '
+                f'value of MEL_MAX_HZ ({self.params.MEL_MAX_HZ}Hz). Upsample'
+                ' your audio stimulus (recommended) or pass a lower value of '
+                'MEL_MAX_HZ when initializing the extractor.')
+
+        model = self.yamnet.yamnet_frames_model(self.params)
+        model.load_weights(self.weights_path)
+        preds, _ = model.predict_on_batch(np.reshape(stim.data, [1,-1]))
+        preds = preds.numpy()[:,self.label_idx]
```

Comment:
Not sure if numpy indexing works with generators; see comment above re: `label_idx`.

Reply:
turned `label_idx` into list (although indexing with generator was no problem)